pkg/server/backend_manager.go

```diff
@@ -105,67 +132,93 @@ type DefaultBackendStorage struct {
 	// Golang.
 	agentIDs []string
 	random   *rand.Rand
+	// idTypes contains the valid identifier types for this
+	// DefaultBackendStorage. The DefaultBackendStorage may only tolerate certain
+	// types of identifiers when associating to a specific BackendManager,
+	// e.g., when associating to the DestHostBackendManager, it can only use the
+	// identifiers of types, IPv4, IPv6 and Host.
+	idTypes []pkgagent.IdentifierType
 }
 
 // NewDefaultBackendManager returns a DefaultBackendManager.
 func NewDefaultBackendManager() *DefaultBackendManager {
-	return &DefaultBackendManager{DefaultBackendStorage: NewDefaultBackendStorage()}
+	return &DefaultBackendManager{
+		DefaultBackendStorage: NewDefaultBackendStorage(
+			[]pkgagent.IdentifierType{pkgagent.UID})}
 }
 
 // NewDefaultBackendStorage returns a DefaultBackendStorage
-func NewDefaultBackendStorage() *DefaultBackendStorage {
+func NewDefaultBackendStorage(idTypes []pkgagent.IdentifierType) *DefaultBackendStorage {
 	return &DefaultBackendStorage{
 		backends: make(map[string][]*backend),
 		random:   rand.New(rand.NewSource(time.Now().UnixNano())),
+		idTypes:  idTypes,
+	}
+}
+
+func containIdType(idTypes []pkgagent.IdentifierType, idType pkgagent.IdentifierType) bool {
+	for _, it := range idTypes {
+		if it == idType {
+			return true
+		}
 	}
+	return false
 }
 
 // AddBackend adds a backend.
-func (s *DefaultBackendStorage) AddBackend(agentID string, conn agent.AgentService_ConnectServer) Backend {
+func (s *DefaultBackendStorage) AddBackend(identifier string, idType pkgagent.IdentifierType, conn agent.AgentService_ConnectServer) Backend {
+	if !containIdType(s.idTypes, idType) {
+		klog.ErrorS(&ErrWrongIDType{idType, s.idTypes}, "fail to add backend")
```

Comment:
This seems an alarming and misleading error to log. As I understand it, it is expected that we would call AddBackend on every IdentifierType an agent presents for each BackendStorage. This means we expect some of those identifiers to not match all BackendStorage and these error message are expected and do not (normally) indicate an actual error. We should either advertise the supported backends and filter this call to prevent this from being expected. Or we should fix the level V(4)? and probably adjust the message to be less alarming "backend not using unsupported type"

Reply:
Will switch to `klog.V(4).Infof`